tmt/utils.py

```diff
@@ -168,15 +170,17 @@ def __str__(self):
 
     @classmethod
     def _save_context(cls, context):
-        """ Save provided command line context for future use """
+        """ Save provided command line context and options for future use """
         cls._context = context
+        cls._options = dict([(key, context.params[key]) for key in
```

Comment:
To iterate over the params dictionary, because `cls._options` is just linked to `context.params`(i.e. they shares the same memory object) if we just assign it via `cls._options = context.params`. e.g.

* 01 - Before invoking `cls._options = context.params`
```python
(Pdb) ll
171  	    @classmethod
172  	    def _save_context(cls, context):
173  	        """ Save provided command line context and options for future use """
174  	        cls._context = context
175  	        import pdb; pdb.set_trace()
176  ->	        cls._options = context.params
177  	        cls._options = dict([(key, context.params[key]) for key in context.params.keys()])
(Pdb) pp id(cls._options), id(context.params)
(139935328340864, 139935324642176)
```
Please be noted that `id(cls._options)` != `id(context.params)`.

---
* 02 - After invoking `cls._options = context.params`
```python
(Pdb) n
> ...<snip>.../tmt/utils.py(177)_save_context()
-> cls._options = dict([(key, context.params[key]) for key in context.params.keys()])
(Pdb) pp id(cls._options), id(context.params)
(139935324642176, 139935324642176)
```
Please be noted that `id(cls._options)` == `id(context.params)`!

---
* 03 - After invoking `cls._options = dict([(key, context.params[key]) for key in ...`
```python
(Pdb) n
> ...<snip>.../tmt/utils.py(177)_save_context()->None
-> cls._options = dict([(key, context.params[key]) for key in context.params.keys()])
(Pdb) pp id(cls._options), id(context.params)
(139935323010816, 139935324642176)
```
Please be noted that `id(cls._options)` != `id(context.params)`.

---
Anyway, I'll use `cls._options = context.params` instead.

Reply:
Yeah, I understand that assigning will share the same object in the memory. My question was rather about why creating a duplicate of the dictionary instead of just reusing the same one (which is shorter, faster and saves memory).

Reply:
OK, got it. Thanks！